gitpod-io/gitpod
# Introduce '/headless-log-download' endpoint for transparent log downloads

Fixes #5084.

~ The code is there, but testing was blocked by some devstaging/imagebuild problems. :confused: ~

Added [this](https://github.com/gitpod-io/gitpod/commit/d24f0f997c1c3d234870b90cb0b533e8c7e2db07) to use the old image-builder: works.

### Test
1. go to `https://gpl-5084-download-logs.staging.gitpod-dev.com/admin/users` and give yourself the "teams-and-projects" permission
1. go to https://gpl-5084-download-logs.staging.gitpod-dev.com/teams/new, link with gitlab and create a new team (if you have no repo with prebuild: create one based on the ruby-on-rails template)
1. go to your new team on the dashboard, go to "Branches" and hit "..." -> "rerun prebuild"
1. wait until the prebuild has finished
2. go to "Prebuilds" and hit "..." -> "view prebuild"
3. see the logs without any CORS issues

**Note:** The original issue with CORS is not testable on devstaging and we should verify on staging. We can, however, test that the change did not introduce a regression.
